homeassistant/components/template/lawn_mower.py

```diff
@@ -0,0 +1,204 @@
+) -> None:
+    """Set up the template lawn mowers."""
+    async_add_entities(await _async_create_entities(hass, config))
+
+
+class TemplateLawnMower(TemplateEntity, LawnMowerEntity):
+    """A template lawn mower component."""
+
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        hass,
+        object_id,
+        config,
+        unique_id,
+    ):
+        """Initialize the lawn mower."""
+        super().__init__(
+            hass, config=config, fallback_name=object_id, unique_id=unique_id
+        )
+        self.entity_id = async_generate_entity_id(
+            ENTITY_ID_FORMAT, object_id, hass=hass
+        )
+        friendly_name = self._attr_name
+
+        self._template = config.get(CONF_VALUE_TEMPLATE)
+
+        self._start_mowing_script = None
+        if start_mowing_action := config.get(SERVICE_START_MOWING):
+            self._start_mowing_script = Script(
+                hass, start_mowing_action, friendly_name, DOMAIN
+            )
+            self._attr_supported_features |= LawnMowerEntityFeature.START_MOWING
+
+        self._pause_script = None
+        if pause_action := config.get(SERVICE_PAUSE):
+            self._pause_script = Script(hass, pause_action, friendly_name, DOMAIN)
+            self._attr_supported_features |= LawnMowerEntityFeature.PAUSE
+
+        self._dock_script = None
+        if dock_action := config.get(SERVICE_DOCK):
+            self._dock_script = Script(hass, dock_action, friendly_name, DOMAIN)
+            self._attr_supported_features |= LawnMowerEntityFeature.DOCK
+
+        self._attr_activity = None
+
+    @property
+    def activity(self) -> LawnMowerActivity | None:
+        """Return the status of the lawn mower."""
+        return self._attr_activity
+
+    async def async_start_mowing(self) -> None:
+        """Start or resume the mowing task."""
+        if self._start_mowing_script is None:
+            return
+        await self.async_run_script(self._start_mowing_script, context=self._context)
+
+    async def async_pause(self) -> None:
+        """Pause the mowing task."""
+        if self._pause_script is None:
+            return
+
+        await self.async_run_script(self._pause_script, context=self._context)
+
+    async def async_dock(self, **kwargs: Any) -> None:
+        """Set the lawn mower to return to the dock."""
+        if self._dock_script is None:
+            return
+
+        await self.async_run_script(self._dock_script, context=self._context)
+
+    @callback
+    def _async_setup_templates(self) -> None:
+        """Set up templates."""
+        if self._template is not None:
+            self.add_template_attribute(
+                "_attr_activity", self._template, None, self._update_state
+            )
+        super()._async_setup_templates()
+
+    @callback
+    def _update_state(self, result):
+        super()._update_state(result)
+        if isinstance(result, TemplateError):
+            # This is legacy behavior
```

Comment:
What does this mean? It's copied from template vacuum obviously, it it still needed in this new platform?

Reply:
Should be sufficient with
```python
if isinstance(result, TemplateError):
    self._state = None
    return
```